dataproc/src/main/java/Quickstart.java

```diff
@@ -0,0 +1,172 @@
+    // settings. The client only needs to be created once and can be reused for multiple requests.
+    // Using a try-with-resources closes the client, but this can also be done manually with
+    // the .close() method.
+    try (ClusterControllerClient clusterControllerClient =
+            ClusterControllerClient.create(clusterControllerSettings);
+        JobControllerClient jobControllerClient =
+            JobControllerClient.create(jobControllerSettings)) {
+      // Configure the settings for our cluster
+      InstanceGroupConfig masterConfig =
+          InstanceGroupConfig.newBuilder()
+              .setMachineTypeUri("n1-standard-1")
+              .setNumInstances(1)
+              .build();
+      InstanceGroupConfig workerConfig =
+          InstanceGroupConfig.newBuilder()
+              .setMachineTypeUri("n1-standard-1")
+              .setNumInstances(2)
+              .build();
+      ClusterConfig clusterConfig =
+          ClusterConfig.newBuilder()
+              .setMasterConfig(masterConfig)
+              .setWorkerConfig(workerConfig)
+              .build();
+      // Create the cluster object with the desired cluster config
+      Cluster cluster =
+          Cluster.newBuilder().setClusterName(clusterName).setConfig(clusterConfig).build();
+
+      // Create the Cloud Dataproc cluster
+      OperationFuture<Cluster, ClusterOperationMetadata> createClusterAsyncRequest =
+          clusterControllerClient.createClusterAsync(projectId, region, cluster);
+      Cluster response = createClusterAsyncRequest.get();
+      System.out.printf("Cluster created successfully: %s", response.getClusterName());
+
+      // Configure the settings for our job
+      JobPlacement jobPlacement = JobPlacement.newBuilder().setClusterName(clusterName).build();
+      PySparkJob pySparkJob = PySparkJob.newBuilder().setMainPythonFileUri(jobFilePath).build();
+      Job job = Job.newBuilder().setPlacement(jobPlacement).setPysparkJob(pySparkJob).build();
+
+      // Submit an asynchronous request to execute the job
+      Job request = jobControllerClient.submitJob(projectId, region, job);
+      String jobId = request.getReference().getJobId();
+      System.out.println(String.format("Submitted job \"%s\"", jobId));
+
+      // Wait for the job to finish
+      CompletableFuture<Job> finishedJobFuture =
+          CompletableFuture.supplyAsync(
+              () -> {
+                // States that indicate a job is "finished"
+                while (true) {
+                  // Poll the service periodically until the Job is in a finished state.
+                  Job jobInfo = jobControllerClient.getJob(projectId, region, jobId);
+                  switch (jobInfo.getStatus().getState()) {
+                    case DONE:
+                      return jobInfo;
+                    case CANCELLED:
+                    case ERROR:
+                    case ATTEMPT_FAILURE:
```

Comment:
ATTEMPT_FAILURE is not a terminal state. Its used to handle failure for restartable jobs, but leads back to RUNNING for another attempt